src/program.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  Keypair,
+  Connection,
+  BpfLoader,
+  BPF_LOADER_PROGRAM_ID,
+  SystemProgram,
+  Transaction,
+  sendAndConfirmTransaction,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import { ethers } from 'ethers';
+import { ContractDeployResult } from 'src';
+
+import { Contract, ContractDeployOptions } from './contract';
+import {
+  LogsParser,
+  LogCallback,
+  EventCallback,
+  parseTxError,
+  parseTxLogs,
+} from './logs';
+
+export class Program {
+  private logs: LogsParser;
+
+  /**
+   * Load a new Solang program
+   *
+   * @param connection    Solana connection
+   * @param payerAccount  Payer pubkey
+   * @param so            Solang program build bundle
+   * @returns             The program
+   */
+  static async deploy(
+    connection: Connection,
+    payerAccount: Keypair,
+    so: Buffer
+  ): Promise<Program> {
+    const programAccount = Keypair.generate();
+
+    await BpfLoader.load(
+      connection,
+      payerAccount,
+      programAccount,
+      so,
+      BPF_LOADER_PROGRAM_ID
+    );
+
+    return new Program(connection, payerAccount, programAccount);
+  }
+
+  /**
+   *
+   * Load a deployed Solang program
+   *
+   * @param connection      Solana connection
+   * @param payerAccount    Payer pubkey
+   * @param programAccount  The program's pubkey
+   * @param so              Solang program build bundle
+   * @returns               The program
+   */
+  static async get(
+    connection: Connection,
+    payerAccount: Keypair,
+    programAccount: Keypair,
+    so: Buffer
+  ): Promise<Program> {
+    await BpfLoader.load(
+      connection,
+      payerAccount,
+      programAccount,
+      so,
+      BPF_LOADER_PROGRAM_ID
+    );
+
+    return new Program(connection, payerAccount, programAccount);
+  }
+
+  /**
+   *
+   * @param connection      Solana connection
+   * @param payerAccount    Payer pubkey
+   * @param programAccount  The program's pubkey
+   */
+  constructor(
+    public connection: Connection,
+    public payerAccount: Keypair,
+    public programAccount: Keypair
+  ) {
+    this.logs = new LogsParser(this.programAccount.publicKey, this.connection);
+  }
+
+  /**
+   * Create a storage account
+   *
+   * @param space
+   * @returns
+   */
+  public async createStorageAccount(space: number): Promise<Keypair> {
+    const lamports = await this.connection.getMinimumBalanceForRentExemption(
+      space
+    );
+
+    let account = Keypair.generate();
```

Comment:
You're right, the account should be caller-provided. In fact, I think it can just be the public key.

Reply:
done. though retained the arg as a keypair since we need to pass the signer here https://github.com/solana-labs/solana-solidity.js/pull/2/files/21087f0919e5dd90e510a0ec02ea474a6bb5964d#diff-34464090febe886c2a40aad4e347f95a9821cff69d4a43a793a376a466b3c885R119